In [23]:
from pathlib import Path
import pandas as pd
import scipy.stats as stats
import matplotlib.pyplot as plt
import datetime
import numpy as np

In [24]:
 # Create a file path 
url = "Resources/chicago_crimes_all_cleaned.csv"

 # Read in the data. 
crime_data = pd.read_csv(url)
crime_data.head()

Case Number                    Date                   Block  \
0    JC104662  12/31/2018 11:59:00 PM  112XX S SACRAMENTO AVE   
1    JC100043  12/31/2018 11:57:00 PM     084XX S SANGAMON ST   
2    JC100006  12/31/2018 11:56:00 PM      018XX S ALLPORT ST   
3    JC100031  12/31/2018 11:55:00 PM     078XX S SANGAMON ST   
4    JC100026  12/31/2018 11:49:00 PM      052XX W GLADYS AVE   

           Primary Type                           Description  District  Ward  \
0  MINOR PROPERTY CRIME                            TO VEHICLE        22  19.0   
1  MINOR PROPERTY CRIME                           TO PROPERTY         6  21.0   
2     ASSAULT & BATTERY  AGG: HANDS/FIST/FEET NO/MINOR INJURY        12  25.0   
3     ASSAULT & BATTERY               DOMESTIC BATTERY SIMPLE         6  17.0   
4     ASSAULT & BATTERY                  AGGRAVATED - HANDGUN        15  29.0   

   Year   Latitude  Longitude                       Location  Population  \
0  2018  41.689079 -87.696064  (41.689078832, -87.696064026)     28804.0   
1  2018  41.740521 -87.647391  (41.740520866, -87.647390719)     68096.0   
2  2018  41.857068 -87.657625  (41.857068095, -87.657625201)     79205.0   
3  2018  41.751914 -87.647717   (41.75191443, -87.647716532)     68096.0   
4  2018  41.875684 -87.760479   (41.87568438, -87.760479356)     47712.0   

   Zipcode  
0  60655.0  
1  60620.0  
2  60608.0  
3  60620.0  
4  60644.0

In [25]:
#calculate average number of crimes per year
avg_crime = crime_data["Case Number"].count()/5
avg_crime

229134.0

In [26]:
#calculating the total count of districts 
counts_of_districts = crime_data["District"].nunique()
counts_of_districts

22

In [27]:
#calculating total crime
total_crime = crime_data["Primary Type"].count()
total_crime

1145670

In [95]:
#calculate the population by district
avg_pop = crime_data.groupby("District")["Population"].mean()
avg_pop

District
1     16836.270024
2     39106.512301
3     46176.181234
4     60732.001801
5     57932.355321
6     65107.393277
7     30013.732689
8     73214.610711
9     56348.951438
10    74752.508318
11    40673.920783
12    38209.084116
14    75317.635411
15    51107.210999
16    49048.371692
17    77549.980608
18    30605.849916
19    69986.150301
20    60738.838983
22    55006.101734
24    47430.891683
25    76863.694187
Name: Population, dtype: float64

In [102]:
#calculate crime count by
crime_count_by_district = crime_data.groupby("District")["Primary Type"].count()
district_avg = crime_count_by_district/5
district_avg

District
1     11933.0
2     10552.4
3     11459.2
4     12768.8
5     10421.0
6     14940.6
7     11989.8
8     13500.0
9     10007.8
10    10724.0
11    15597.6
12    12016.8
14     7840.6
15     7898.6
16     7588.0
17     6198.4
18    12203.8
19    10674.6
20     4383.4
22     7161.8
24     7389.4
25    11884.4
Name: Primary Type, dtype: float64

In [104]:
district_per_cap = district_avg/avg_pop
district_per_cap

District
1     0.708767
2     0.269837
3     0.248163
4     0.210248
5     0.179882
6     0.229476
7     0.399477
8     0.184389
9     0.177604
10    0.143460
11    0.383479
12    0.314501
14    0.104100
15    0.154550
16    0.154704
17    0.079928
18    0.398741
19    0.152524
20    0.072168
22    0.130200
24    0.155793
25    0.154617
dtype: float64

In [29]:
#expected value
expected_total = total_crime/22
expected_total

52075.90909090909

In [30]:
# Observed counts of crime by district (cumulative, over the last 5 years)
observed = pd.Series([59665.00000000000,52762.00000000000,57296.00000000000,63844.00000000000,52105.00000000000,74703.00000000000,59949.00000000000,67500.00000000000,50039.00000000000,53620.00000000000,77988.00000000000,60084.00000000000,39203.00000000000,39493.00000000000,37940.00000000000,30992.00000000000,61019.00000000000,53373.00000000000,21917.00000000000,35809.00000000000,36947.00000000000,59422.00000000000], 
                     index=['1','2','3','4','5','6','7','8','9','10','11','12','14','15','16','17','18','19','20','22','24','25'])

In [31]:
# Create a data frame
df = pd.DataFrame([observed]).T

In [32]:
# Add a column whose default values are the expected values
df[1] = 52075.90909090909

In [33]:
# Rename columns
df.columns = ["observed", "expected"]

In [34]:
# View the data frame
df 

observed      expected
1    59665.0  52075.909091
2    52762.0  52075.909091
3    57296.0  52075.909091
4    63844.0  52075.909091
5    52105.0  52075.909091
6    74703.0  52075.909091
7    59949.0  52075.909091
8    67500.0  52075.909091
9    50039.0  52075.909091
10   53620.0  52075.909091
11   77988.0  52075.909091
12   60084.0  52075.909091
14   39203.0  52075.909091
15   39493.0  52075.909091
16   37940.0  52075.909091
17   30992.0  52075.909091
18   61019.0  52075.909091
19   53373.0  52075.909091
20   21917.0  52075.909091
22   35809.0  52075.909091
24   36947.0  52075.909091
25   59422.0  52075.909091

In [35]:
# The degree of freedom is 22-1 = 21
# With a p-value of 0.05, the confidence level is 1.00-0.05 = 0.95.
critical_value = stats.chi2.ppf(q = 0.95, df = 21)
# The critical value
critical_value 

32.670573340917315

In [36]:
# Run the chi square test with stats.chisquare()
stats.chisquare(df['observed'], df['expected']) 

Power_divergenceResult(statistic=82280.93870311696, pvalue=0.0)

Conclusion

Since the chi square value of 82,281 exceeds the critical value of 32.67, we conclude that the results are statistically significant.

In [37]:
#get an array of all years
years = crime_data["Year"].unique()

In [38]:
# Split data file into years
df_2018 = crime_data[crime_data["Year"]==2018]
df_2019 = crime_data[crime_data["Year"]==2019]
df_2020 = crime_data[crime_data["Year"]==2020]
df_2021 = crime_data[crime_data["Year"]==2021]
df_2022 = crime_data[crime_data["Year"]==2022]
df_2018

Case Number                    Date                   Block  \
0         JC104662  12/31/2018 11:59:00 PM  112XX S SACRAMENTO AVE   
1         JC100043  12/31/2018 11:57:00 PM     084XX S SANGAMON ST   
2         JC100006  12/31/2018 11:56:00 PM      018XX S ALLPORT ST   
3         JC100031  12/31/2018 11:55:00 PM     078XX S SANGAMON ST   
4         JC100026  12/31/2018 11:49:00 PM      052XX W GLADYS AVE   
...            ...                     ...                     ...   
894540    JB115033  01/02/2018 11:00:00 AM      064XX W ADDISON ST   
894541    JB101356  01/02/2018 08:05:00 AM    051XX N NORMANDY AVE   
894542    JB101127  01/01/2018 10:30:00 PM    062XX W CORNELIA AVE   
894543    JB100762  01/01/2018 12:00:00 PM    063XX W WAVELAND AVE   
894544    JB104090  01/01/2018 11:30:00 AM        100XX W OHARE ST   

                Primary Type                           Description  District  \
0       MINOR PROPERTY CRIME                            TO VEHICLE        22   
1       MINOR PROPERTY CRIME                           TO PROPERTY         6   
2          ASSAULT & BATTERY  AGG: HANDS/FIST/FEET NO/MINOR INJURY        12   
3          ASSAULT & BATTERY               DOMESTIC BATTERY SIMPLE         6   
4          ASSAULT & BATTERY                  AGGRAVATED - HANDGUN        15   
...                      ...                                   ...       ...   
894540     THEFT (ALL TYPES)                             OVER $500        16   
894541     ASSAULT & BATTERY                                SIMPLE        16   
894542    OTHER MINOR CRIMES        HARASSMENT BY ELECTRONIC MEANS        16   
894543    OTHER MINOR CRIMES        HARASSMENT BY ELECTRONIC MEANS        16   
894544     THEFT (ALL TYPES)                             OVER $500        16   

        Ward  Year   Latitude  Longitude                       Location  \
0       19.0  2018  41.689079 -87.696064  (41.689078832, -87.696064026)   
1       21.0  2018  41.740521 -87.647391  (41.740520866, -87.647390719)   
2       25.0  2018  41.857068 -87.657625  (41.857068095, -87.657625201)   
3       17.0  2018  41.751914 -87.647717   (41.75191443, -87.647716532)   
4       29.0  2018  41.875684 -87.760479   (41.87568438, -87.760479356)   
...      ...   ...        ...        ...                            ...   
894540  36.0  2018  41.945654 -87.786853  (41.945653748, -87.786852586)   
894541  41.0  2018  41.974559 -87.793957  (41.974559221, -87.793956917)   
894542  36.0  2018  41.943873 -87.782380  (41.943873325, -87.782379691)   
894543  38.0  2018  41.947501 -87.784668  (41.947501166, -87.784667989)   
894544  41.0  2018  41.976290 -87.905227  (41.976290414, -87.905227221)   

        Population  Zipcode  
0          28804.0  60655.0  
1          68096.0  60620.0  
2          79205.0  60608.0  
3          68096.0  60620.0  
4          47712.0  60644.0  
...            ...      ...  
894540     24272.0  60706.0  
894541     24272.0  60706.0  
894542     24272.0  60706.0  
894543     24272.0  60706.0  
894544     20694.0  60106.0  

[258966 rows x 13 columns]

In [108]:
# Create a data frame of population from year to year, noting that zip codes repeat, thus creating repeated population counts,
# since we pulled population based on zip code. So, we groupby unique zip code, get the population for each zip code, and sum by population,
# then by population across all zip codes for one year
population = [df_2018.groupby("Zipcode")["Population"].unique().sum().sum(),
              df_2019.groupby("Zipcode")["Population"].unique().sum().sum(),
              df_2020.groupby("Zipcode")["Population"].unique().sum().sum(),
              df_2021.groupby("Zipcode")["Population"].unique().sum().sum(),
              df_2022.groupby("Zipcode")["Population"].unique().sum().sum()]
population = np.array(population)
population.mean()

3005790.6

In [40]:
# Look at types of crime per capita on the year level
# Count total crimes by crime type per year
year_crime_cnt = crime_data.groupby("Year")["Primary Type"].count()
year_crime_cnt

# Put it in a dataframe
crime_by_year = pd.DataFrame({"Year" : year_crime_cnt.index, "Total Crime" : year_crime_cnt.values})
crime_by_year

Year  Total Crime
0  2018       258966
1  2019       254518
2  2020       204271
3  2021       199060
4  2022       228855

In [41]:
#total number of unique crime
crime_2018 = df_2018["Primary Type"].count()
crime_2018

258966

In [42]:
#crime by district
crime_by_district_2018 = df_2018.groupby("District")["Primary Type"].count()
crime_by_district_2018

District
1     15249
2     11608
3     11926
4     13569
5     11775
6     16272
7     14124
8     15203
9     10874
10    12480
11    18817
12    12818
14     9151
15     8778
16     7945
17     7201
18    15829
19    11930
20     4836
22     7887
24     7546
25    13148
Name: Primary Type, dtype: int64

In [43]:
#expected value
expected_2018 = crime_2018/22
expected_2018

11771.181818181818

In [44]:
# Observed counts of crime by district
observed_2018 = pd.Series([15249.000000000000,11608.000000000000,11926.000000000000,13569.000000000000,11775.000000000000,16272.000000000000,14124.000000000000,15203.000000000000,10874.000000000000,12480.000000000000,18817.000000000000,12818.000000000000,9151.000000000000,8778.000000000000,7945.000000000000,7201.000000000000,15829.000000000000,11930.000000000000,4836.000000000000,7887.000000000000,7546.000000000000,13148.000000000000], 
                     index=['1','2','3','4','5','6','7','8','9','10','11','12','14','15','16','17','18','19','20','22','24','25'])

In [45]:
# Create a data frame
chi_2018 = pd.DataFrame([observed_2018]).T

In [46]:
 # Add a column whose default values are the expected values
chi_2018[1] = 11771.181818181818

In [47]:
# Rename columns
chi_2018.columns = ["observed", "expected"]

In [48]:
# View the data frame
chi_2018

observed      expected
1    15249.0  11771.181818
2    11608.0  11771.181818
3    11926.0  11771.181818
4    13569.0  11771.181818
5    11775.0  11771.181818
6    16272.0  11771.181818
7    14124.0  11771.181818
8    15203.0  11771.181818
9    10874.0  11771.181818
10   12480.0  11771.181818
11   18817.0  11771.181818
12   12818.0  11771.181818
14    9151.0  11771.181818
15    8778.0  11771.181818
16    7945.0  11771.181818
17    7201.0  11771.181818
18   15829.0  11771.181818
19   11930.0  11771.181818
20    4836.0  11771.181818
22    7887.0  11771.181818
24    7546.0  11771.181818
25   13148.0  11771.181818

In [49]:
# Run the chi square test with stats.chisquare()
stats.chisquare(chi_2018['observed'], chi_2018['expected'])

Power_divergenceResult(statistic=21728.357884818855, pvalue=0.0)

Conclusion

Since the chi square value of 21,728 exceeds the critical value of 32.67, we conclude that the results are statistically significant.

In [50]:
#total number of unique crime
crime_2019 = df_2019["Primary Type"].count()
crime_2019

254518

In [51]:
#crime by district
crime_by_district_2019 = df_2019.groupby("District")["Primary Type"].count()
crime_by_district_2019

District
1     15002
2     11072
3     12404
4     13518
5     11274
6     16755
7     13674
8     14598
9     10934
10    12413
11    18597
12    12960
14     9106
15     8867
16     7710
17     6440
18    14919
19    11531
20     4284
22     7745
24     7874
25    12841
Name: Primary Type, dtype: int64

In [52]:
#expected value
expected_2019 = crime_2019/22
expected_2019

11569.0

In [53]:
# Observed counts of crime by district
observed_2019 = pd.Series([15002, 11072,12404,13518,11274,16755,13674,14598,10934,12413,18597,12960,9106,8867,7710,6440,14919,11531,4284,7745,7874,12841], 
                     index=['1','2','3','4','5','6','7','8','9','10','11','12','14','15','16','17','18','19','20','22','24','25'])

In [54]:
# Create a data frame
chi_2019 = pd.DataFrame([observed_2019]).T

In [55]:
 # Add a column whose default values are the expected values
chi_2019[1] = 11569

In [56]:
# Rename columns
chi_2019.columns = ["observed", "expected"]

In [57]:
# View the data frame
chi_2019

observed  expected
1      15002     11569
2      11072     11569
3      12404     11569
4      13518     11569
5      11274     11569
6      16755     11569
7      13674     11569
8      14598     11569
9      10934     11569
10     12413     11569
11     18597     11569
12     12960     11569
14      9106     11569
15      8867     11569
16      7710     11569
17      6440     11569
18     14919     11569
19     11531     11569
20      4284     11569
22      7745     11569
24      7874     11569
25     12841     11569

In [58]:
# Run the chi square test with stats.chisquare()
stats.chisquare(chi_2019['observed'], chi_2019['expected'])

Power_divergenceResult(statistic=22328.108393119543, pvalue=0.0)

Conclusion

Since the chi square value of 22,328 exceeds the critical value of 32.67, we conclude that the results are statistically significant.

In [59]:
#total number of unique crime
crime_2020 = df_2020["Primary Type"].count()
crime_2020

204271

In [60]:
#crime by district
crime_by_district_2020 = df_2020.groupby("District")["Primary Type"].count()
crime_by_district_2020

District
1      8150
2      9270
3     10869
4     11711
5     10067
6     13902
7     11648
8     12178
9      9105
10     9994
11    14780
12     9886
14     6455
15     7813
16     7024
17     5566
18     8555
19     8977
20     3930
22     6628
24     6621
25    11142
Name: Primary Type, dtype: int64

In [61]:
#expected value
expected_2020 = crime_2020/22
expected_2020

9285.045454545454

In [62]:
# Observed counts of crime by district
observed_2020 = pd.Series([8150.000000000000,9270.000000000000,10869.000000000000,11711.000000000000,10067.000000000000,13902.000000000000,11648.000000000000,12178.000000000000,9105.000000000000,9994.000000000000,14780.000000000000,9886.000000000000,6455.000000000000,7813.000000000000,7024.000000000000,5566.000000000000,8555.000000000000,8977.000000000000,3930.000000000000,6628.000000000000,6621.000000000000,11142.000000000000], 
                     index=['1','2','3','4','5','6','7','8','9','10','11','12','14','15','16','17','18','19','20','22','24','25'])

In [63]:
# Create a data frame
chi_2020 = pd.DataFrame([observed_2020]).T

In [64]:
 # Add a column whose default values are the expected values
chi_2020[1] = 9285.045454545454

In [65]:
# Rename columns
chi_2020.columns = ["observed", "expected"]

In [66]:
# View the data frame
chi_2020

observed     expected
1     8150.0  9285.045455
2     9270.0  9285.045455
3    10869.0  9285.045455
4    11711.0  9285.045455
5    10067.0  9285.045455
6    13902.0  9285.045455
7    11648.0  9285.045455
8    12178.0  9285.045455
9     9105.0  9285.045455
10    9994.0  9285.045455
11   14780.0  9285.045455
12    9886.0  9285.045455
14    6455.0  9285.045455
15    7813.0  9285.045455
16    7024.0  9285.045455
17    5566.0  9285.045455
18    8555.0  9285.045455
19    8977.0  9285.045455
20    3930.0  9285.045455
22    6628.0  9285.045455
24    6621.0  9285.045455
25   11142.0  9285.045455

In [67]:
 # Run the chi square test with stats.chisquare()
stats.chisquare(chi_2020['observed'], chi_2020['expected'])

Power_divergenceResult(statistic=16443.998007548795, pvalue=0.0)

Conclusion

Since the chi square value of 16,443 exceeds the critical value of 32.67, we conclude that the results are statistically significant.

In [68]:
#total number of unique crime
crime_2021 = df_2021["Primary Type"].count()
crime_2021

199060

In [69]:
#crime by district
crime_by_district_2021 = df_2021.groupby("District")["Primary Type"].count()
crime_by_district_2021

District
1      8768
2      9313
3     10350
4     11805
5      9395
6     13307
7     10339
8     11845
9      8949
10     8938
11    12906
12    10506
14     6618
15     6869
16     7097
17     5315
18     9716
19     9216
20     4073
22     6473
24     6660
25    10602
Name: Primary Type, dtype: int64

In [70]:
#expected value
expected_2021 = crime_2021/22
expected_2021

9048.181818181818

In [71]:
# Observed counts of crime by district
observed_2021 = pd.Series([8768.000000000000,9313.000000000000,10350.000000000000,11805.000000000000,9395.000000000000,13307.000000000000,10339.000000000000,11845.000000000000,8949.000000000000,8938.000000000000,12906.000000000000,10506.000000000000,6618.000000000000,6869.000000000000,7097.000000000000,5315.000000000000,9716.000000000000,9216.000000000000,4073.000000000000,6473.000000000000,6660.000000000000,10602.000000000000], 
                     index=['1','2','3','4','5','6','7','8','9','10','11','12','14','15','16','17','18','19','20','22','24','25'])

In [72]:
# Create a data frame
chi_2021 = pd.DataFrame([observed_2021]).T

In [73]:
 # Add a column whose default values are the expected values
chi_2021[1] = 9048.181818181818

In [74]:
# Rename columns
chi_2021.columns = ["observed", "expected"]

In [75]:
# View the data frame
chi_2021

observed     expected
1     8768.0  9048.181818
2     9313.0  9048.181818
3    10350.0  9048.181818
4    11805.0  9048.181818
5     9395.0  9048.181818
6    13307.0  9048.181818
7    10339.0  9048.181818
8    11845.0  9048.181818
9     8949.0  9048.181818
10    8938.0  9048.181818
11   12906.0  9048.181818
12   10506.0  9048.181818
14    6618.0  9048.181818
15    6869.0  9048.181818
16    7097.0  9048.181818
17    5315.0  9048.181818
18    9716.0  9048.181818
19    9216.0  9048.181818
20    4073.0  9048.181818
22    6473.0  9048.181818
24    6660.0  9048.181818
25   10602.0  9048.181818

In [76]:
 # Run the chi square test with stats.chisquare()
stats.chisquare(chi_2021['observed'], chi_2021['expected'])

Power_divergenceResult(statistic=13549.006611072038, pvalue=0.0)

Conclusion

Since the chi square value of 13,549 exceeds the critical value of 32.67, we conclude that the results are statistically significant.

In [77]:
#total number of unique crime
crime_2022 = df_2022["Primary Type"].count()
crime_2022

228855

In [78]:
#crime by district
crime_by_district_2022 = df_2022.groupby("District")["Primary Type"].count()
crime_by_district_2022

District
1     12496
2     11499
3     11747
4     13241
5      9594
6     14467
7     10164
8     13676
9     10177
10     9795
11    12888
12    13914
14     7873
15     7166
16     8164
17     6470
18    12000
19    11719
20     4794
22     7076
24     8246
25    11689
Name: Primary Type, dtype: int64

In [79]:
#expected value
expected_2022 = crime_2022/22
expected_2022

10402.5

In [80]:
# Observed counts of crime by district
observed_2022 = pd.Series([12496.0,11499.0,11747.0,13241.0,9594.0,14467.0,10164.0,13676.0,10177.0,9795.0,12888.0,13914.0,7873.0,7166.0,8164.0,6470.0,12000.0,11719.0,4794.0,7076.0,8246.0,11689.0], 
                     index=['1','2','3','4','5','6','7','8','9','10','11','12','14','15','16','17','18','19','20','22','24','25'])

In [81]:
# Create a data frame
chi_2022 = pd.DataFrame([observed_2022]).T

In [82]:
 # Add a column whose default values are the expected values
chi_2022[1] = 10402.5

In [83]:
# Rename columns
chi_2022.columns = ["observed", "expected"]

In [84]:
# View the data frame
chi_2022

observed  expected
1    12496.0   10402.5
2    11499.0   10402.5
3    11747.0   10402.5
4    13241.0   10402.5
5     9594.0   10402.5
6    14467.0   10402.5
7    10164.0   10402.5
8    13676.0   10402.5
9    10177.0   10402.5
10    9795.0   10402.5
11   12888.0   10402.5
12   13914.0   10402.5
14    7873.0   10402.5
15    7166.0   10402.5
16    8164.0   10402.5
17    6470.0   10402.5
18   12000.0   10402.5
19   11719.0   10402.5
20    4794.0   10402.5
22    7076.0   10402.5
24    8246.0   10402.5
25   11689.0   10402.5

In [85]:
 # Run the chi square test with stats.chisquare()
stats.chisquare(chi_2022['observed'], chi_2022['expected'])

Power_divergenceResult(statistic=14687.330112953618, pvalue=0.0)

Conclusion

Since the chi square value of 14,687 exceeds the critical value of 32.67, we conclude that the results are statistically significant.

In [86]:
# Look at per capita crime rate per district over the five year timespan
district_df = crime_data[["Year", "District", "Primary Type", "Population", "Zipcode"]]
district_df

# Get the population per district by creating a new data frame and dropping any unique combinations of distict, zip code, and population
unique_pop = district_df[["District", "Zipcode", "Population"]].drop_duplicates()
pop_per_district = unique_pop.groupby("District")["Population"].sum()

# Get the per capita crime count
cnt_per_district = district_df.groupby("District")["Primary Type"].count()
cnt_per_district

# Get the per capita crime count
per_cap_crime_cnt = cnt_per_district/pop_per_district

# Put in a data frame
district_per_capita = pd.DataFrame({"District" : per_cap_crime_cnt.index, "Per Capita Crime Count" : per_cap_crime_cnt.values})
district_per_capita
np.array(pop_per_district).sum()

27922342.0

In [87]:
pop_per_district

District
1      612009.0
2      930963.0
3     1025051.0
4      950580.0
5     1408567.0
6     1116104.0
7      754255.0
8     1572497.0
9     2071502.0
10    1753354.0
11    1280221.0
12    1455686.0
14    1563365.0
15     799152.0
16    1578312.0
17    1434298.0
18     807249.0
19    2096604.0
20    1149048.0
22     780412.0
24     690298.0
25    2092815.0
Name: Population, dtype: float64

In [88]:
total_capita = district_per_capita["Per Capita Crime Count"].sum()
total_capita

1.0285460386540908

In [89]:
# Observed counts of crime by district (cumulative, over the last 5 years)
per_capita_observed = pd.Series([0.097490000000000000,0.056675000000000000,0.055896000000000000,0.067163000000000000,0.036991000000000000,0.066932000000000000,0.079481000000000000,0.042925000000000000,0.024156000000000000,0.030581000000000000,0.060918000000000000,0.041275000000000000,0.025076000000000000,0.049419000000000000,0.024038000000000000,0.021608000000000000,0.075589000000000000,0.025457000000000000,0.019074000000000000,0.045885000000000000,0.053523000000000000,0.028393000000000000], 
                     index=['1','2','3','4','5','6','7','8','9','10','11','12','14','15','16','17','18','19','20','22','24','25'])

In [111]:
expected_per_capita = (avg_crime/(population.mean()))/ 22
expected_per_capita

0.0034650390543445766

In [118]:
# Create a data frame
chi_per_capita = pd.DataFrame([district_per_cap]).T

In [113]:
# Add a column whose default values are the expected values
chi_per_capita[1] = expected_per_capita

In [114]:
# Rename columns
chi_per_capita.columns = ["observed", "expected"]

In [115]:
# View the data frame
chi_per_capita 

observed  expected
District                    
1         0.708767  0.003465
2         0.269837  0.003465
3         0.248163  0.003465
4         0.210248  0.003465
5         0.179882  0.003465
6         0.229476  0.003465
7         0.399477  0.003465
8         0.184389  0.003465
9         0.177604  0.003465
10        0.143460  0.003465
11        0.383479  0.003465
12        0.314501  0.003465
14        0.104100  0.003465
15        0.154550  0.003465
16        0.154704  0.003465
17        0.079928  0.003465
18        0.398741  0.003465
19        0.152524  0.003465
20        0.072168  0.003465
22        0.130200  0.003465
24        0.155793  0.003465
25        0.154617  0.003465

In [116]:
# Run the chi square test with stats.chisquare()
stats.chisquare(chi_per_capita['observed'], chi_per_capita['expected']) 

Power_divergenceResult(statistic=444.6848511922491, pvalue=5.002199887979559e-81)

Conclusion

Since the chi square value of 0.21 is below the critical value of 32.67, we conclude that the results are statistically significant.